chassis.go

```diff
@@ -134,12 +155,64 @@ func (odbi *ovndb) chassisGetImp(hostname string) ([]*Chassis, error) {
 }
 
 func (odbi *ovndb) rowToChassis(uuid string) (*Chassis, error) {
+
+	cacheChassis, ok := odbi.cache[tableChassis][uuid]
+	if !ok {
+		return nil, fmt.Errorf("Chassis with uuid%s not found", uuid)
+	}
 	ch := &Chassis{
 		UUID:       uuid,
-		Name:       odbi.cache[tableChassis][uuid].Fields["name"].(string),
-		Hostname:   odbi.cache[tableChassis][uuid].Fields["hostname"].(string),
-		ExternalID: odbi.cache[tableChassis][uuid].Fields["external_ids"].(libovsdb.OvsMap).GoMap,
-		NbCfg:      odbi.cache[tableChassis][uuid].Fields["nb_cfg"].(int),
+		Name:       cacheChassis.Fields["name"].(string),
+		Hostname:   cacheChassis.Fields["hostname"].(string),
+		ExternalID: cacheChassis.Fields["external_ids"].(libovsdb.OvsMap).GoMap,
+		NbCfg:      cacheChassis.Fields["nb_cfg"].(int),
+	}
+
+	if tz, ok := cacheChassis.Fields["transport_zones"]; ok {
+		switch tz.(type) {
+		case libovsdb.UUID:
+			ch.TransportZones = []string{tz.(libovsdb.UUID).GoUUID}
+		case libovsdb.OvsSet:
+			ch.TransportZones = odbi.ConvertGoSetToStringArray(tz.(libovsdb.OvsSet))
+		}
+	}
+	if vtep, ok := cacheChassis.Fields["vtep_logical_switches"]; ok {
+		switch vtep.(type) {
+		case libovsdb.UUID:
+			ch.VtepLogicalSwitches = []string{vtep.(libovsdb.UUID).GoUUID}
+		case libovsdb.OvsSet:
+			ch.VtepLogicalSwitches = odbi.ConvertGoSetToStringArray(vtep.(libovsdb.OvsSet))
+		}
+	}
+	var encaps []string
+	if enc, ok := cacheChassis.Fields["encaps"]; ok {
+		switch enc.(type) {
+		case libovsdb.UUID:
+			if enuid, ok := enc.(libovsdb.UUID); ok {
+				cenc, err := odbi.rowToEncap(enuid.GoUUID)
```

Comment:
what's the purpose of this rowToEncap()? You convert from UUID to Encap, but later, finally, you just append the UUID to encaps, then why do this conversion at all for the UUID of Encap?

Reply:
You are right. No need of rowToEncap. Taken care in revised commit